rmc-docs/src/tutorial/arbitrary-variables/src/inventory.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+// ANCHOR: inventory_lib
+use std::num::NonZeroU32;
+use vector_map::VecMap;
+
+type ProductId = u32;
+
+pub struct Inventory {
+    inner: VecMap<ProductId, NonZeroU32>,
+}
+
+impl Inventory {
+    pub fn update(&mut self, id: ProductId, new_quantity: NonZeroU32) {
+        self.inner.insert(id, new_quantity);
+    }
+
+    pub fn get(&self, id: &ProductId) -> Option<NonZeroU32> {
+        self.inner.get(id).cloned()
+    }
+}
+// ANCHOR_END: inventory_lib
+
+#[cfg(rmc)]
+mod verification {
+    use super::*;
+
+    // ANCHOR: safe_update
+    #[rmc::proof]
+    pub fn safe_update() {
+        // Create inventory variable.
+        let mut inventory = Inventory { inner: VecMap::new() };
```

Comment:
So ideally we'd use `any` but we don't support that for heap types yet, correct?

Reply:
Yes. We don't have an implementation for VecMap today.